### DISCUSSION TITLE
(Questions) Multiple questions and "possible" suggestions/bug reports?

### DISCUSSION THREAD

**Discussion by AZFire1** [Zrips/CMI#4286] [Q&A]
**Question 1:**
I've noticed with some commands CMI just sends back a whole bunch of dashes like 
"----------------------------------------------------------- (Yellow)". 
One of the commands this happens for is /cmi checkban. Is this supposed to happen, and why?

**Question 2:**
Does CMI have it's own uptime placeholder, or way to edit the uptime message/use it in other plugins (the placeholder). If there is no way to edit uptime message and there is no uptime placeholder, maybe this could be added at some point. Not needed though.

**Question 3:**
So, I've converted from essentials economy. I have a towny server and I am using the CMI Custom compiled version of Vault.. When I changed economy plugins all the economy of my servers towns reset.. How do I re-do this without this happening again?

**Question 4:**
Can you make elevators with blocks and jumping? For example, you can put a diamond block above another diamond block and jump and you'd get moved up to the next block? If not, could this maybe be added at some point, it's not that important though..

**Comment by CloudeLecaw** [comment]
Question 1: its up on your serverand font (ressourcepack) your using.
With normal vanilla default all looksfine in chat. often it helps to descipe or design ingame

Question 2: You define the meassage in langfile for own meassage else where its recomment to use placeholderAPI with extansion server (/papi ecloud download Server) because this can be used everywhere and the update from it is equal to Cmis.
the placeholder i suggest is using %server_uptime%

For editing the style of /cmi status u fine it in langfile:
```
  status:
    help:
      info: '&eShow server status'
      args: ''
    info:
      Platform: ' &ePlatform: &6[platform] &e(&6[os]&e) &eRunning threads: &6[threads]'
      Tps: ' &eTPS: &6[tps]'
      CpuUsage: ' &eCpu usage: &6[cpuusage]% &e(&6[cores] cores&e)'
      Uptime:
        main: ' &eUptime: '
        weeks: '&e[weeks] &6weeks '
        days: '&e[days] &6days '
        hours: '&e[hours] &6hours '
        mins: '&e[mins] &6mins '
        secs: '&e[secs] &6secs '
      MemoryUsage: ' &eMemory usage: &6[percent]% &e(&6[current]&e/&6[max] MB&e)'
      MemoryUsageHover: "&fFree: &7[free] \n&fAllocated: &7[allocated] \n&fMax: &7[max]\
        \ \n&fTotal: &7[total]"
      Java: ' &eJava version: &6[mainV].[majorV].[minorV]&e(&6[update]&e) Build: &6[build]'
      DiskUsage: ' &eDisk usage: &6[percent]% &e(&6[current]&e/&6[max] GB&e)'
      Worlds: '&e [number]. &6[world] &e[chunks] Chunks &6[entities] Entities &e[tiles]
        tiles &6[players] players'
```

Question3:
have you tried the migration command from console?
**cmi migratedatabase
Changes database system and migrates all data**

Question4: at the moment this is not how elevators work. if u wish that i suggest youwrite a little request ticket for this :)
Well may you mean the "animated flying" some elevatorplugins does when moving up an down?
[see here how lift plugins make movement](https://youtu.be/k-IT_ULndbE?t=132)

**Comment by AZFire1** [comment]
Would I need to re-install essentials, or add back the essentials file and then use the cmi migratedatabase command?

**Comment by CloudeLecaw** [comment]
You got it ;) they need to run both until Migration is done

**Comment by AZFire1** [comment]
So uh, my server crashed with a ton of errors..

**Comment by CloudeLecaw** [comment]
On Migration?

**Comment by AZFire1** [comment]
Yes

**Comment by CloudeLecaw** [comment]
So what have you done wrong, or what have u triwd? It also would be helpfull to have a log to see what went wrong :3

**Comment by AZFire1** [comment]
The server stopped responding until it crashed. I rolled back to a backup of the server I took before.. CMI was saying it had no mysql database to convert to..

**Comment by CloudeLecaw** [comment]
This means you gesetet up mysql in database.yml
Check the database. Yml what you have setet up.
If using sqslite then dont use cmi migrate command. 

Dont use this if your not using mysql. By default cmi uses sqlite. This command is only usefull if you have some data allready and you wanne changes from sqslite to mysql. 
```
migratedatabase
Changes database system and migrates all data

```
I know its called migrate but what you need to do is Import Informationen from somewhere else :) 
I aspect you used this command without any msql Informationen inside the database Informationen yml and your server have up with a endless error in log because he cant Connect to mysql. 
Anyway you dont neednthis command if your doing a fresh setup and also dont wanne change to mysql. 

 Use this first
>cmi importoldusers
> Imports users from playerdata folder in main world folder. Server can suffer lag spike during import

And after that import Informationen you want from essentials
> cmi importfrom [essentials/hd] [home/warp/nick/logoutlocation/money/mail]
> Import data from other plugins like essentials or holographicdisplays

So this is what you can taken from essentials
Homes, warpa, nicknames, logoutlocations, mails and money (if using economy dont forget you need vault + cmiinjector plugin or the modified vault version from cmi wiki)
Use command once for each thing u want to import from essentials.

**Comment by AZFire1** [comment]
I am using SQLite.. I've already done all this.. That's how I got to where I am now.. I could try it again though.

**Comment by CloudeLecaw** [comment]
delet the old db fileand in cmi try aggain.
if importing was sucessfullthen then run the server without Essentials to see if all was imported to your needs

[if something went wrong it would be nice to have a copy of your latest.log here.](https://hastebin.com/) Its hard to tell what went wrong when the only information is "the server crashed or hang up" Sometimes it shows information what went wrong. (for example yesterday there was a user with a pluginloader for PluginByWorld  that was causing theproblem) Without the log we cant get an idea what may went wrong because we currently know nothing about your server. That is what the log is for, to give information and inform. So please post the log if it wents wrong on next try

**Comment by AZFire1** [comment]
Alright, well I've created a test server and started a "test" town with essentials and deposited money into it (the server only had essentials, cmi, towny, & vault). After I added town money to my town, I added CMI and changed to it's Vault. I imported from essentials (with economy enabled, & changed the currency symbol set to "$" if that matters) and no matter how many times I imported it, the town balance simply didn't transfer after I changed from essentials to cmi (especially after importing). Does CMI not import all money data from essentials?
I went to Towny's discord. The people there said that all the money is stored in the economy plugin..
![image](https://user-images.githubusercontent.com/67762583/94842097-f739fa00-03e8-11eb-93e4-53e1ba596c74.png)

(Note: It does import the player balances though doesn't import other balances (such as towny).)


Here's all the logs with me testing.. They are all simple, and short due to the fact I only have 3 - 4 plugins on this server.
1 - https://mcpaste.io/1fa5d115f1137c13
2 - https://mcpaste.io/ad8bfbf11b2fe9d5
3 - https://mcpaste.io/2da6742517d6fe44
4 - https://mcpaste.io/cca2d3690d53f9e0
5 - https://mcpaste.io/4105914220dff7b7
6 - https://mcpaste.io/417c88096338d730

Just as a recap, I'd like to point out everything I did in the process (to make it a little less confusing):
 - Add essentials & towny & vault.
 - Start server.
 - Give myself and my town a lot of money through essentials economy.
 - Stop server.
 - Add CMI.
 - Start server.
 - "/cmi importoldusers" and "/cmi importfrom essentials money" commands executed.
 - Stop server.
 - Enable economy through CMI & add it's Vault (remove old vault). (Also, as I mentioned above change the currency sign, as I didn't know if that could maybe help)
 - Remove essentials.
 - Start server.
 - Notice everything hooks up correctly with towny & cmi..
 - Do "/bal", notice you have the correct balance.
 - Look at your town balance, ("/town" or "/t") notice it is wiped & reset..
 - Try importing from essentials again though nothing changes..

Sorry for being a little stressed and aggressive about this, I'm just kind of annoyed as I can't get it to work correctly :/

**Comment by CloudeLecaw** [comment]
It does. Do you have CMIInjecotr installed?

Economy: For CMI economy to properly work you will need one of two solutions: 
1. Is to use recompiled Vault version from [HERE](http://www.zrips.net/wp-content/uploads/2020/07/Vault-1.7.3.jar) which has direct support for CMI economy and will have best compatibility with other plugins. 
2. Use economy injector from [HERE](http://www.zrips.net/wp-content/uploads/2020/07/CMIEInjector1.0.2.3.jar) which will add CMI economy support to any Vault build.
Both methods are valid but second one can have small issues with plugins which would tend to load straight after Vault loads and before injector himself. 
P.S. Don’t forget to enable Economy in config file.

Since the last version of Zrips vault sometimes
